### DISCUSSION TITLE
Xiaomi Robot Vacuum X20+

### DISCUSSION THREAD

**Discussion by alx6ths** [Tasshack/dreame-vacuum#746] [Q&A] [answered]
Hi. We recently bought an X20+ and after some time searching I found information yesterday that there is support for it in beta version 2.0.0b15. Subsequently, I discovered that there is also a newer version 16, but it doesn't work with the robot and I had to switch back to 15. However, the map does not appear in that one.
I don't know where the problem could be, could someone help me please?

Thank you.

p.s. I also tried Xiaomi Cloud Map Extractor but it doesn't support X20+...

**Comment by Tasshack** [comment]
https://github.com/Tasshack/dreame-vacuum/issues/498

**Reply by Tasshack** [reply]
It should work with v2.0.0.b16

**Reply by alx6ths** [reply]
Thank you very much for the recommendation, but as I wrote, nothing works in version v2.0.0.b16 and in version v2.0.0.b15 at least data, but no map.
I don't currently have Home Assistant Core up to date, could this be the problem?
Thanks again and sorry to bother you with this.

![Screenshot 2024-10-07 130627](https://github.com/user-attachments/assets/76f4dfdf-09bd-42ba-9d06-cce4b95f0f9c)
![Screenshot 2024-10-07 132514](https://github.com/user-attachments/assets/1bc1894a-d890-41cb-ad6d-5f49a0f285db)
![Screenshot 2024-10-07 132536](https://github.com/user-attachments/assets/6e81f29f-ee39-4c66-8961-82a5da84af51)

**Reply by Tasshack** [reply, marked as answer]
> I don't currently have Home Assistant Core up to date, could this be the problem?

Probably thats why because the HA device info does not have a serial_number variable that I have added on b16.

**Reply by alx6ths** [reply]
Ok, thank you. I needed to confirm this before going ahead with the update. The last time I did a Core update, some other issues showed up that will require more complex fixes. That's why I haven't done it yet...

**Reply by Tasshack** [reply]
Or you could just delete this line.

https://github.com/Tasshack/dreame-vacuum/blob/dev/custom_components/dreame_vacuum/entity.py#L179

**Reply by alx6ths** [reply]
Great, it's working now! Thank you very much!
Although I could have the opportunity to finally update HS and fix the errors that appeared in Multiscrape then :-D But I'll definitely get to that too, thanks again.

**Reply by alx6ths** [reply]
Hi. After finally getting around to updating to the latest version of HA Core, thanks for the kick, **I can confirm** that v2.0.0.b16 works even without the recommended entity.py edit.

Thank you once again, also for the fact that thanks to this I finally decided to do a problematic update, because it meant fixing modbus.yaml where there are dozens of sensors that I use to load from the air conditioning system and from the gas boiler).
Thank God that even after updating other components that followed the HA update, everything works as it should.
